addons/crm_sale_project/wizard/project_template_create_wizard.py

```diff
@@ -0,0 +1,39 @@
+from odoo import api, fields, models
+
+
+class ProjectTemplateCreateWizard(models.TransientModel):
+    _inherit = 'project.template.create.wizard'
+
+    customer_action = fields.Selection([
+        ('create', 'Create a new customer'),
+        ('exist', 'Link to an existing customer'),
+        ('nothing', 'Do not link to a customer'),
+    ], string='Customer')
+
+    @api.model
+    def action_open_template_view(self):
+        action = super().action_open_template_view()
+        if self.env.context.get("default_lead_id"):
+            action['context']['default_lead_id'] = self.env.context.get("default_lead_id")
+        return action
+
+    def action_create_project_from_lead(self):
+        """Create a project either from template or directly if no template is set."""
+        self.ensure_one()
+        lead_id = self.env['crm.lead'].browse(self.env.context.get("default_lead_id"))
+
+        if self.customer_action == 'create':
+            lead_id._handle_partner_assignment(create_missing=True)
+            self.partner_id = lead_id.partner_id
+        elif self.customer_action == 'exist':
+            lead_id._handle_partner_assignment(force_partner_id=self.partner_id.id, create_missing=False)
+
+        if self.template_id:
+            project = self._create_project_from_template()
+        else:
+            values = {
+                'partner_id': self.partner_id.id,
+                'company_id': lead_id.company_id.id,
+            }
+            project = self.env['project.project'].create(values)
+        return project.action_view_tasks()
```

Comment:
Usually, for a variable name, we add `_id` as suffix in variable name when this variable contains an id of a record but if the variable should contain a recordset then we don't suffix the `_id`. 😇 

```suggestion
        lead = self.env['crm.lead'].browse(self.env.context.get("default_lead_id"))

        if self.customer_action == 'create':
            lead._handle_partner_assignment(create_missing=True)
            self.partner_id = lead.partner_id
        elif self.customer_action == 'exist':
            lead._handle_partner_assignment(force_partner_id=self.partner_id.id, create_missing=False)

        if self.template_id:
            project = self._create_project_from_template()
        else:
            values = {
                'partner_id': self.partner_id.id,
                'company_id': lead.company_id.id,
            }
            project = self.env['project.project'].create(values)
        return project.action_view_tasks()
```